Use transaction for complex db inserts

maciekglowka opened the pull request:
Initial solution for the second part of #675 

Refactor of the `Database` type: internal methods moved to stand-alone functions, that take db connection as an argument. The functions are now generic over the `ConnectionTrait`, as this seems to be the convention that SeaOrm follows as well.
`Provider` implementation methods that make multiple changes to the DB have been wrapped in a transaction, to assure data consistency on failure.
Tests have been updated.

secana commented:
Could you please rebase the PR against the "devel" branch and format it with the default "cargo fmt". That should make the PR much smaller.

maciekglowka commented:
Hi again. I've rebased to devel and changed the PR target to the branch as well (as this seems to be the procedure now, according to the contributing guide).
Indeed I had `edition="2021" in my `rustfmt.toml` - hence the differences. But I've changed it to a blank default now and reformatted.

I am afraid github's differ doesn't handle well the total refactor of the `database.rs` file. The diff view is quite a mess :/ 
Anyway I've applied the differences between main and devel (mostly token hashing) - so it should be now fully compatible with the current devel branch. (initialy I worked on main). At least the tests seem to pass locally :)
